docs/CNS.md

```diff
@@ -3,100 +3,123 @@
 
-```
-# CHANGE THE NEXT LINE TO POINT TO THE LOCATION OF THE CNSsolve DIRECTORY
-
-        CNS_SOLVE=/PATH/TO/cns_solve_1.3
+```bash
+sed -i 's@_CNSsolve_location_@'"$PWD"'@' cns_solve_env
 ```
 
-## 2 Add HADDOCK routines
+> **Note:** On a Mac, the default sed version will give an error with the above command. Install instead `gnu-sed` using `brew`, and then replace `sed` by `gsed` in the above command.
 
-HADDOCK requires some adjustments in CNS to function properly.
-The set of routines to add is provided in the haddock3 repository under [`varia/cns1.3/`](/varia/cns1.3/README.md).
-Copy all of these files into the CNS source directory (`/PATH/TO/cns_solve_1.3/source/`), for example:
+Check if the substitution worked:
 
 ```bash
-cp ~/software/haddock3/varia/cns1.3/* source/
+sed -n 18p cns_solve_env
 ```
 
-## 3 Compile CNS
+## 3. Patch CNS with HADDOCK custom files
+
+For HADDOCK to use CNS, some modifications were made to the CNS source code. We provide those modifications together with the haddock3 source files.
 
-Make sure you are inside the main `cns_solve_1.3` directory.
+> ❗ You cannot use CNS in HADDOCK without these patched files❗
 
-If a suitable compiler is installed on your system, the following command should start the compilation of CNS:
+Copy the CNS files distributed with HADDOCK3 to the CNS directory:
 
 ```bash
-make install
+\cp -r ~/software/haddock3/varia/cns1.3/[bis]*  ~/software/cns_solve_1.3/
 ```
 
-If needed, you can specify the compiler you want to use by adding the `compiler` option, for example:
+> **Note:** Make sure to use the correct path to your haddock3 location in the above command.
 
-```bash
-make install compiler=gfortran
-```
+## 4. Compiling CNS
 
-You should see the progress of the compilation, followed by a confirmation that an executable has been created. For example (with gfortran):
+Before installing CNS, you need to check wether your operating system is supported by CNS.
 
-```
-created executable file cns_solve-2206031450.exe
+The list of supported architectures can be seen with the following command:
+
+```bash
+ls ~/software/cns_solve_1.3/instlib/machine/supported
+arm-aarch64-linux  intel-x86_64bit-linux  linux  mac-arm-darwin  mac-intel-darwin
 ```
 
-If this does not work, check the [troubleshooting section](#Troubleshooting) below.
+To check what is *your* architectures, type the following command:
 
-## 4 Source CNS
+```bash
+~/software/cns_solve_1.3/bin/getarch
+```
 
-Lastly, CNS needs to be loaded in the current shell before it can be used.
+> You might get something like `unknown-...`, in that case installation is still possible, however it means the compiler options might not be properly defined - check the Troubleshooting section
 
-For csh/tcsh:
+Several `Makefile` headers are already provided - here we will focus on the ones that use `gfortran` as the compiler.
 
 ```bash
-source cns_solve_env
+ls ~/software/cns_solve_1.3/instlib/machine/supported/*/Makefile*gfortran*
```

Comment:
Is this really necessary? 
Don’t see a use of listing makefiles, unless user is instructed to check that a specific file is present etc